krm-functions/scripts/release.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+#
+# Copyright 2022 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Usage:
+#   ./release.sh $FUNCTION_NAME
+#   ./release.sh $FUNCTION_NAME [patch|minor|major]
+#
+# If the provided function has never been released before, this script will
+# release v0.1.0.
+#
+# Otherwise, it will do a semver bump based on the second argument (default is patch
+# if no second argument is provided). This will push a tag $FUNCTION_NAME/$VERSION
+# to the Github repo and will push to the release branch, e.g. $FUNCTION_NAME/v0.2.
+#
+# $FUNCTION_NAME should be of the form $PUBLISHER/$FUNCTION
+#
+# After computing the correct version number to release, the script will ask the
+# user to confirm the release version and the function name before triggering
+# the release.
+#
+# This script assumes that the remote Github repository to push the branch/tag
+# is called "upstream".
+
+set -e
+
+REMOTE="upstream"
+
+if [ "$#" -lt 1 ]; then
+    echo "function name required as first parameter"
+    exit 1
+fi
+
+FUNCTION_NAME=$1
+
+# Check that the function source code actually exists in this repository.
+for d in */ ; do
+  if [ $d != "scripts/" ]; then
+    for e in "$d"*/ ; do
+      fn_name=${e%/}
+      if [ $fn_name = $FUNCTION_NAME ]; then
+        found=true
+      fi
+    done
+  fi
+done
+
+if [ -z "$found" ]; then
+    echo "error: source code for $FUNCTION_NAME not found in this repo"
+    exit 1
+fi
+
+# If no release type is set, default to patch.
+RELEASE_TYPE=$2
+if [ -z "$RELEASE_TYPE" ]; then
+    RELEASE_TYPE="patch"
+fi
+
+# Get the most recently released tag of this function.
+LAST_TAG=`curl -sL http://api.github.com/repos/kubernetes-sigs/krm-functions-registry/releases | jq ".[].tag_name" | grep $FUNCTION_NAME | head -n 1`
```

Comment:
I don't think it's guaranteed, the documentation doesn't say anything about the order. From my experiences with using this api, it has always looked like it is sorted by the date that the release was published. But, since that's not guaranteed, I think you're right that it would be better to sort and search. I'll make that change.